File: src/main/java/go/scala/model/Entry.java
package go.scala.model;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

@Entity
public class Entry {
	
	@Id
	@GeneratedValue
	private Long id;
	private String title;
	private String text;
	private boolean published;
	
	@Temporal(TemporalType.TIMESTAMP)
	private Date created;
	
	@OneToMany(cascade=CascadeType.ALL)
	private List<Comment> comments = new ArrayList<Comment>();
	
	public Long getId() {
		return id;
	}
	public void setId(Long id) {
		this.id = id;
	}
	public String getTitle() {
		return title;
	}
	public void setTitle(String title) {
		this.title = title;
	}
	public String getText() {
		return text;
	}
	public void setText(String text) {
		this.text = text;
	}
	public boolean isPublished() {
		return published;
	}
	public void setPublished(boolean published) {
		this.published = published;
	}
	public List<Comment> getComments() {
		return comments;
	}
	public void setComments(List<Comment> comments) {
		this.comments = comments;
	}
	public Date getCreated() {
		return created;
	}
	public void setCreated(Date created) {
		this.created = created;
	}
	public void addComment(Comment comment) {
		comments.add(comment);
	}

}

File: src/main/scala/go/scala/model/Entry.scala
package go.scala.model

import java.util.{ArrayList, Date, List}

import javax.persistence.{CascadeType, Entity, GeneratedValue, Id, OneToMany, Temporal, TemporalType}

import reflect.BeanInfo

@Entity
@BeanInfo
class Entry {
	
	@Id
	@GeneratedValue
	var id:Long = _
	var title:String = _
	var text:String = _
	var published:Boolean = _
	
	@Temporal(TemporalType.TIMESTAMP)
	var created:Date = _
	
	@OneToMany(cascade = Array(CascadeType.ALL))
	var comments:List[Comment] = new ArrayList

	def addComment(comment:Comment) {
		comments.add(comment)
	}
	
	override def hashCode = {
		val prime = 31;
		var result = 1;
		prime * result + (id match{
			case l:Long => l.hashCode
			case _ => 0
		})
	}
	
	override def equals(other:Any) = {
		other match {
			case e:Entry if(id == e.id) => true
			case _ => false
		}
	}

	override def toString() = {
		"Entry [comments=" + comments + ", created=" + created + ", id=" + id + 
			", published=" + published + ", text=" + text + ", title=" + title + "]"
	}

}
